package NYT.NTableClient.NProto;

import "chunk.proto";

////////////////////////////////////////////////////////////////////////////////

message TRange
{
    required bytes Begin = 1;
    required bytes End = 2;
    required bool IsInfinite = 3;
}

message TBlockInfo
{
    required int32 BlockIndex = 1;
    required int32 RowCount = 2;
}

message TChannel 
{
    repeated bytes Columns = 1;
    repeated TRange Ranges = 2;
    repeated TBlockInfo Blocks = 3;
}

message TTableChunkAttributes
{
    extend NYT.NChunkServer.NProto.TChunkAttributes
    {
        required TTableChunkAttributes TableAttributes = 101;
    }

    repeated TChannel Channels = 1;
    required int32 CodecId = 2;
    required int64 RowCount = 3;
}

////////////////////////////////////////////////////////////////////////////////

